### PR TITLE
Add summary and done message

### LINKED ISSUE #92: Add action summary and "done" message to log output.
**What I'd like:**
Updater currently reports successful updates as they occur. This means that larger clusters will have the results of their updates spread throughout the logs. This can make it difficult to determine if all the update attempts were successful. Additionally it is unclear from the logs if actions are still being undertaken, or if Updater is done. 

As such, a summary of events should be printed to the logs at the end of its run along with a message clearly indicating that all actions are complete until the next run. 

**Any alternatives you've considered:**
N/A

### PR DESCRIPTION
**Pull request by WilboMo**
**Issue number:**
Closes #92 

**Description of changes:**
Adds a summary of actions taken during the update process and
a message indicating when all operations have been completed.

**Testing done:**
Ran `make test`: 
```
PASS
ok      github.com/bottlerocket-os/bottlerocket-ecs-updater     (cached)
```

Executed changes against test cluster containing a mix of AL2 and Bottlerocket instances along with a non-service task running. Summary output from test run: 

```
2021/07/22 17:27:02 After action summary:
2021/07/22 17:27:02 i-003951b31d363ab7e: Instance updated successfully
2021/07/22 17:27:02 i-034c9d175a47eb509: Instance updated successfully
2021/07/22 17:27:02 i-0a9f1d0e5f5c56d9b: Instance is not eligible for updates because it contains non-service task(s)
2021/07/22 17:27:02 Update operations complete!
```
Above instances were destroyed after testing. 

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: updater/main.go [hunk 1]
```diff
@@ -144,11 +148,14 @@ func _main() error {
 		activateErr := u.activateInstance(i.containerInstanceID)
 		if updateErr != nil && activateErr != nil {
 			log.Printf("Failed to update instance %#q: %v", i, updateErr)
+			summary[i.instanceID] = fmt.Sprintf("Failed to update and failed reactivate. Update error: %v, Aactivation error: %v", updateErr, activateErr)
```
Comment (srgothi92) [inline updater/main.go]: Updater will immediately stop after the return, therefore summary will not be printed and there is no point in additng it to summary.
```suggestion
			
```

File: updater/main.go [hunk 2]
```diff
@@ -144,11 +148,14 @@ func _main() error {
...
 			return fmt.Errorf("instance %#q failed to re-activate after failing to update: %w", i, activateErr)
 		} else if updateErr != nil {
 			log.Printf("Failed to update instance %#q: %v", i, updateErr)
+			summary[i.instanceID] = fmt.Sprintf("Failed to update: %v", updateErr)
 			continue
 		} else if activateErr != nil {
+			summary[i.instanceID] = fmt.Sprintf("Failed to re-activate after update: %v", activateErr)
```
Comment (srgothi92) [inline updater/main.go]: Updater will stop after return, same as above comment.
```suggestion
			
```

File: updater/main.go [hunk 3]
```diff
@@ -158,13 +165,21 @@ func _main() error {
...
 		if err != nil {
 			log.Printf("Failed to verify update for instance %#q: %v", i, err)
+			summary[i.instanceID] = fmt.Sprintf("Failed to verify update: %v", err)
 		}
 		if !ok {
 			log.Printf("Update failed for instance %#q", i)
+			summary[i.instanceID] = "Update failed"
 		} else {
 			log.Printf("Instance %#q updated successfully!", i)
+			summary[i.instanceID] = "Instance updated successfully"
```
Comment (srgothi92) [inline updater/main.go]: I think we should only have one line of summary for each instance. All the conditions here are not mutually exclusive and would end up replacing other. I think below should be okay in summary.
```suggestion
			
		}
		if !ok {
			log.Printf("Update failed for instance %#q", i)
			summary[i.instanceID] = "Update failed"
		} else {
			log.Printf("Instance %#q updated successfully!", i)
			summary[i.instanceID] = "Instance updated successfully"
```

File: updater/main.go (hunk 1 above)
Comment (WilboMo) [reply updater/main.go]: That's right, somehow that slipped my mind. Thanks for the call out!

**Review by srgothi92 — APPROVED** [review]

**Review by samuelkarp — APPROVED** [review]

**Pull request merged**